proto/data.go

```diff
@@ -316,6 +317,69 @@ func (v *Value) SetProto(msg gogoproto.Message) error {
 	return nil
 }
 
+// SetTime encodes the specified time value into the bytes field of the
+// receiver and sets the tag.
+func (v *Value) SetTime(t time.Time) error {
+	var err error
+	if v.Bytes, err = t.MarshalBinary(); err != nil {
```

Comment:
Interesting point. We'll eventually need to either:
- plumb a custom time zone into this method (see discussion in #2565 about #2327)
- set the custom time zone in the caller, which means we should _not_ muck with the time zone in this method.

On the other hand, assuming we handle time conversions correctly everywhere else, the offset we store doesn't actually matter, and we should just pick the fastest option (which is probably the current code).